rateslib/curves.py

```diff
@@ -306,6 +306,8 @@ def __init__(
         **kwargs,
     ):
         self.id = uuid4().hex[:5] + "_" if id is NoInput.blank else id  # 1 in a million clash
+        if sorted(nodes.keys()) != list(nodes.keys()):
+            raise ValueError("Nodes keys are not sorted, to sort your Curve nodes use: dict(sorted(nodes.items()))")
```

Comment:
Not that this is a performance bottleneck, but by sorting the nodes to do a comparison you have already expended the processing required to do the sort, which is O(n log n), and you are then checking each item in turn which is even more costly.

A more performant solution is to loop through until you find a node that is less than or equal to the last, in which case you can short-circuit to the Error:

```
# check node ordering
for i in range(1, len(node_dates)):
    if node_dates[i] <= node_dates[i-1]:
        raise ValueError(...)
```

Reply:
That is right, I completely forgot about the most critical part of the nodes sorting. Thanks!